extension/js/common/api/account-server.ts

```diff
@@ -2,55 +2,46 @@
+import { ExternalService } from './account-servers/external-service.js';
 import { ParsedRecipients } from './email-provider/email-provider-api.js';
-import { BackendAuthErr } from './shared/api-error.js';
 import { Api, ProgressCb } from './shared/api.js';
+import { ClientConfigurationJson } from '../client-configuration.js';
+import { SHARED_TENANT_API_HOST } from '../core/const.js';
 
 export type UploadedMessageData = {
   url: string; // both FES and FlowCryptComApi
   externalId?: string; // legacy FES
   emailToExternalIdAndUrl?: { [email: string]: { url: string; externalId: string } }; // FES only
 };
 
+export type AccountGetAndUpdateResult = {
+  clientConfiguration: ClientConfigurationJson;
+  defaultWebPortalMessageExpire: number;
+};
+
 /**
  * This may be calling to FlowCryptComApi or Enterprise Server (FES, customer on-prem) depending on
  *   whether FES is deployed on the customer domain or not.
  */
 export class AccountServer extends Api {
+  private readonly potentialCustomUrlFes: ExternalService;
+  private readonly sharedTenantFes: ExternalService;
+
   public constructor(private acctEmail: string) {
     super();
+    this.potentialCustomUrlFes = new ExternalService(this.acctEmail);
+    this.sharedTenantFes = new ExternalService(this.acctEmail);
+    this.sharedTenantFes.url = SHARED_TENANT_API_HOST;
   }
 
-  public accountGetAndUpdateLocalStore = async (): Promise<BackendRes.FcAccountGet> => {
-    if (await this.isFesUsed()) {
-      const fes = new EnterpriseServer(this.acctEmail);
-      const fetchedClientConfiguration = await fes.fetchAndSaveClientConfiguration();
-      /* eslint-disable @typescript-eslint/naming-convention */
-      return {
-        domain_org_rules: fetchedClientConfiguration,
-        // todo - rethink this. On FES, expiration is handled with S3 bucket policy regardless of this number
-        //  which is set to 180 days on buckets we manage. This number below may still be rendered somewhere
-        //  when composing, which should be evaluated.
-        account: { default_message_expire: 180 },
-      };
-      /* eslint-enable @typescript-eslint/naming-convention */
-    } else {
-      return await FlowCryptComApi.accountGetAndUpdateLocalStore(await this.getIdToken());
-    }
-  };
-
-  public accountUpdate = async (profileUpdate: ProfileUpdate): Promise<void> => {
-    if (await this.isFesUsed()) {
-      const fes = new EnterpriseServer(this.acctEmail);
-      await fes.accountUpdate(profileUpdate);
-    } else {
-      await FlowCryptComApi.accountUpdate(await this.getIdToken(), profileUpdate);
-    }
+  public accountGetAndUpdateLocalStore = async (): Promise<AccountGetAndUpdateResult> => {
+    const service = await this.getExternalService();
+    const fetchedClientConfiguration = await service.fetchAndSaveClientConfiguration();
+    return {
+      clientConfiguration: fetchedClientConfiguration,
+      defaultWebPortalMessageExpire: (await this.isFesUsed()) ? 180 : 90,
```

Comment:
Looking at this now, `defaultWebPortalMessageExpire` no longer has to be here, either. I'd remove it from here, and instead move it to where it's actually used. If it's more than one place where it's used, then there should be a method somewhere called `getWebPortalMessageExpireDays` which will either return 180 or 90. The method could possibly be on this class, for now. 

Also `AccountGetAndUpdateResult` could be removed, this method can be called `fetchAndSaveClientConfiguration`  as well, instead of `accountGetAndUpdateLocalStore`, and it can return `ClientConfigurationJson` directly.

Reply:
Sounds good.